gcovr/tests/test_gcovr.py

```diff
@@ -182,7 +184,13 @@ def pytest_generate_tests(metafunc):
                     reason="branch coverage details seem to be platform-dependent"),
                 pytest.mark.xfail(
                     name == 'rounding' and is_windows,
-                    reason="branch coverage seem to be platform-dependent")
+                    reason="branch coverage seem to be platform-dependent"),
+                pytest.mark.xfail(
+                    name == 'html-source-encoding-cp1252' and (os.getenv('CC') == 'gcc-8'),
+                    reason="g++-8 does not support command line option '--encoding'"),
+                pytest.mark.xfail(
+                    name == 'html-source-encoding-cp1252' and (os.getenv('TEST_OPTS') is not None),  # Environment is set in Docker build
+                    reason="g++-5 in Docker does not support command line option '--encoding'")
```

Comment:
I'm not sure why `--encoding` ever worked. According to the [GCC docs](https://gcc.gnu.org/onlinedocs/gcc-5.5.0/gcc/Preprocessor-Options.html#Preprocessor-Options) the correct option should probably be `-finput-charset=CHARSET`.

In any case, I don't think `TEST_OPTS` is an indicator of a Docker environment? It could be set in any build that uses the Makefile.

Reply:
I've searched how to detect the docher container but there is no easy way. I'll remove the xfail and change the option in the tests.

Reply:
Done.